syft/pkg/cataloger/python/parse_poetry_lock.go

```diff
@@ -93,15 +60,31 @@ func parsePoetryLock(_ context.Context, _ file.Resolver, _ *generic.Environment,
 }
 
 func poetryLockPackages(reader file.LocationReadCloser) ([]pkg.Package, error) {
-	tree, err := toml.LoadReader(reader)
+	metadata := poetryPackages{}
+	md, err := toml.NewDecoder(reader).Decode(&metadata)
 	if err != nil {
-		return nil, fmt.Errorf("unable to load poetry.lock for parsing: %w", err)
+		return nil, fmt.Errorf("failed to read poetry lock package: %w", err)
 	}
 
-	metadata := poetryPackages{}
-	err = tree.Unmarshal(&metadata)
-	if err != nil {
-		return nil, fmt.Errorf("unable to parse poetry.lock: %w", err)
+	for i, p := range metadata.Packages {
+		dependencies := make(map[string][]poetryPackageDependency)
+		for pkgName, du := range p.DependenciesUnmarshal {
+			var (
+				single    string
+				singleObj poetryPackageDependency
+				multiObj  []poetryPackageDependency
+			)
+
+			switch {
```

Comment:
if we reach the end of this switch, and no action was taken (no case hit) then we dropped a field. I thikn we should trace log that

Reply:
👍 you're right let me add that